glib/src/gobject/binding_group.rs

```diff
@@ -140,19 +140,26 @@ impl<'a> BindingGroupBuilder<'a> {
             user_data: ffi::gpointer,
         ) -> ffi::gboolean {
             let transform_data =
-                &*(user_data as *const (TransformFn, TransformFn, ParamSpec, ParamSpec));
+                &*(user_data as *const (TransformFn, TransformFn, String, ParamSpec));
+            let binding = from_glib_borrow(binding);
 
             match (transform_data.1.as_ref().unwrap())(
-                &from_glib_borrow(binding),
+                &binding,
                 &*(from_value as *const Value),
             ) {
                 None => false,
                 Some(res) => {
+                    let pspec_name = transform_data.2.clone();
+                    let source = binding.source().unwrap();
+                    let pspec = source.find_property(&pspec_name);
+                    assert!(pspec.is_some(), "Source object does not have a property {}", pspec_name);
+                    let pspec = pspec.unwrap();
+
                     assert!(
-                        res.type_().is_a(transform_data.2.value_type()),
+                        crate::Value::type_transformable(res.type_(), pspec.value_type()),
```

Comment:
This should stay as before. We don't want to loosen the type checks, as discussed.

Reply:
Done.